ee/modules/030-cloud-provider-openstack/hooks/discover.go

```diff
@@ -0,0 +1,198 @@
+/*
+Copyright 2023 Flant JSC
+Licensed under the Deckhouse Platform Enterprise Edition (EE) license. See https://github.com/deckhouse/deckhouse/blob/main/ee/LICENSE
+*/
+
+package hooks
+
+import (
+	"fmt"
+	"regexp"
+	"sort"
+	"strings"
+	"unicode"
+
+	"github.com/flant/addon-operator/pkg/module_manager/go_hook"
+	"github.com/flant/addon-operator/sdk"
+	"github.com/flant/shell-operator/pkg/kube_events_manager/types"
+	log "github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	storage "k8s.io/api/storage/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/util/json"
+
+	"github.com/deckhouse/deckhouse/dhctl/pkg/config"
+	"github.com/deckhouse/deckhouse/go_lib/cloud-data/apis/v1alpha1"
+)
+
+var _ = sdk.RegisterFunc(&go_hook.HookConfig{
+	OnBeforeHelm: &go_hook.OrderedConfig{Order: 20},
+	Kubernetes: []go_hook.KubernetesConfig{
+		{
+			Name:       "cloud_provider_discovery_data",
+			ApiVersion: "v1",
+			Kind:       "Secret",
+			NamespaceSelector: &types.NamespaceSelector{
+				NameSelector: &types.NameSelector{
+					MatchNames: []string{"kube-system"},
+				},
+			},
+			NameSelector: &types.NameSelector{
+				MatchNames: []string{"d8-cloud-provider-discovery-data"},
+			},
+			FilterFunc: applyCloudProviderDiscoveryDataSecretFilter,
+		},
+		{
+			Name:       "storage_classes",
+			ApiVersion: "storage.k8s.io/v1",
+			Kind:       "StorageClass",
+			FilterFunc: applyStorageClassFilter,
```

Comment:
I forget about one thing. Cluster admin can add own storage class. We need to filter storage class by label selector `heritage: deckhouse` as [here](https://github.com/deckhouse/deckhouse/blob/468ae94c1d944d253b2d62a5e26050d1efa0a823/modules/340-monitoring-kubernetes/hooks/storage_class_cloud_manual.go#L61)

Reply:
and add test for cover this case